expose-metrics/handler.go

```diff
@@ -0,0 +1,111 @@
+package function
+
+import (
+	"encoding/json"
+	"fmt"
+	"log"
+	"net/http"
+	"net/url"
+	"os"
+	"regexp"
+	"strconv"
+
+	"github.com/openfaas/faas/gateway/metrics"
+)
+
+type Metrics struct {
+	Success int `json:"success"`
+	Failure int `json:"failure"`
+}
+
+// Handle a serverless request
+func Handle(req []byte) string {
+	fnName, err := parseFunctionName(string(req))
+	if err != nil {
+		log.Fatalf("Couldn't parse query %b, %t", req, err)
+	}
+
+	host := os.Getenv("prometheus_host")
+	envPort := os.Getenv("prometheus_port")
+	port, err := strconv.Atoi(envPort)
+	if err != nil {
+		log.Fatalf("Could not convert env-var prometheus_port to int. Env-var value: %s. Error: %t", envPort, err)
+	}
+
+	metricsQuery := metrics.NewPrometheusQuery(host, port, &http.Client{})
+
+	fnMetrics, err := getMetrics(fnName, metricsQuery)
+	if err != nil {
+		log.Fatalf("Couldn't get metrics from Prometheus for function %s, %t", fnName, err)
+	}
+
+	res, err := json.Marshal(fnMetrics)
+	if err != nil {
+		log.Fatalf("Couldn't marshal json %t", err)
+	}
+	return string(res)
+}
+
+func parseFunctionName(query string) (string, error) {
+	var fnName string
+
+	matcher := regexp.MustCompile("^/?(?:/system/)??(?:async-)?function/([^/?]+)([^?]*)")
+	parsed, err := url.Parse(query)
+	if err != nil {
+		return "", fmt.Errorf("Couldn't parse url %s, %t", query, err)
+	}
+	nameIndex := 1
+
+	matches := matcher.FindStringSubmatch(parsed.Path)
+	if len(matches) >= nameIndex+1 {
+		fnName = matches[nameIndex]
+	}
+
+	return fnName, nil
+}
+
+func getMetrics(fnName string, metricsQuery metrics.PrometheusQueryFetcher) (*Metrics, error) {
+
+	expr := url.QueryEscape(fmt.Sprintf(`sum(increase(gateway_function_invocation_total{function_name="%s"}[60m])) by (code)`, fnName))
```

Comment:
I would make the `60m` a parameter in this function that's configurable via an env var. The default value can be `60m` if no env var is supplied.

Reply:
How would you name the env var? `metrics_time_period`, `metrics_interval`?

Reply:
`metrics_window`